### PR TITLE
Bugfixes

### PR DESCRIPTION
**Pull request by eanders**
## Description

Fixes (but needs thorough QA):
* On live reports, if you choose only homeless project types, it lost ES after the first attempt (regression from FY2024 changes)
* Bugfix for Core Demographics where a client might be missing (usually just dev data)
* LSA kick-off page was checking SO for inventory related issues.  LSA doesn't work on SO, so limit checks to project types that are relevant.
* Some LSA checks were not checking Project Type 0
* Standardize checks for LSA to project types the filter knows about
* State-level reporting was trying to use the new race codes, but the census is unaware of them, which made it unhappy.
* Demographic Summary Excel download included the incorrect title. 

## Type of change
- [x] Bug fix
- [ ] New feature (adds functionality)
- [ ] Code clean-up / housekeeping
- [ ] Dependency update

## Checklist before requesting review
- [ ] I have performed a self-review of my code
- [ ] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [ ] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [ ] My code follows the style guidelines of this project (rubocop)
- [ ] I have updated the documentation (or not applicable)
- [ ] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: app/controllers/admin/talentlms_controller.rb [hunk 1]
```diff
@@ -9,18 +9,32 @@ class TalentlmsController < ApplicationController
...
 
     def index
-      @config = Talentlms::Config.first_or_initialize
+      @config = config_scope.first_or_initialize
+      redirect_to new_admin_talentlm_path if @config.new_record?
+      redirect_to edit_admin_talentlm_path(@config) unless @config.new_record?
+    end
+
+    def new
+      @config = config_scope.first_or_initialize
```
Comment (awisema) [inline app/controllers/admin/talentlms_controller.rb]: I think the use of `first_or_initialize` in `new` and then `create` in `create`, could be confusing... If `new` gets triggered when there is already a config, it will add another unreachable config, and then appear to lose the configuration as the first config is displayed...

File: drivers/hmis/app/graphql/types/base_audit_event.rb [hunk 2]
```diff
@@ -46,11 +46,16 @@ def record_name
     end
 
     def user
+      return unless object.whodunnit
       # 'unauthenticated' matches user_for_paper_trail in ApplicationController.
       # This happens when a Job updates records, which we should display as System changes.
       return User.system_user if object.whodunnit == 'unauthenticated'
 
-      Hmis::User.find_by(id: object.whodunnit)
+      # If user was impersonating, return the true user only. If someone performed the action while impersonating, whodunnit stores as '1 as 2' where 1 is the true user.
```
Comment (awisema) [inline drivers/hmis/app/graphql/types/base_audit_event.rb]: I can't parse this comment, maybe rephrase?

File: drivers/hmis/app/models/hmis/filter/application_user_filter.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+###
+
+class Hmis::Filter::ApplicationUserFilter < Hmis::Filter::BaseFilter
+  def filter_scope(scope)
+    scope = ensure_scope(scope)
+    scope = with_search_term(scope)
+    scope
+  end
+
+  protected
```
Comment (awisema) [inline drivers/hmis/app/models/hmis/filter/application_user_filter.rb]: Move modifiers to methods instead of the section format?

File: drivers/hmis/app/models/hmis/hud/custom_client_address.rb
Comment (awisema) [inline drivers/hmis/app/models/hmis/hud/custom_client_address.rb:1]: Philosophical question: Should the HUD module contain non-HUD defined models?

**Review by awisema — APPROVED** [review]
I'm pretty sure all of the comments are from other PRs that have been pulled into this branch, but I include them anyway :-)

**Comment by eanders** [comment]
> I'm pretty sure all of the comments are from other PRs that have been pulled into this branch, but I include them anyway :-)

Oh, looks like I had the wrong base branch!  Sorry, adjusted.

**Review by gigxz — APPROVED** [review]
